app/src/main/java/com/infomaniak/drive/ui/fileList/SelectFolderActivity.kt

```diff
@@ -57,26 +57,34 @@ class SelectFolderActivity : BaseActivity() {
         val customArgs = navigationArgs.customArgs
         val currentFolderId = navigationArgs.folderId.getIntOrNull()
         val disabledFolderId = navigationArgs.disabledFolderId.getIntOrNull()
-        val currentUserDrive = UserDrive(userId, driveId)
+        DriveInfosController.getDrive(driveId = driveId, maintenance = false)?.let { currentDrive ->
+            val currentUserDrive = UserDrive(userId, driveId, currentDrive.sharedWithMe)
 
-        mainViewModel.selectFolderUserDrive = currentUserDrive
+            mainViewModel.selectFolderUserDrive = currentUserDrive
 
-        selectFolderViewModel.apply {
-            userDrive = currentUserDrive
-            currentDrive = DriveInfosController.getDrive(userId, driveId)
-            disableSelectedFolderId = disabledFolderId
-        }
-
-        setSaveButton(customArgs)
+            selectFolderViewModel.apply {
+                userDrive = currentUserDrive
+                this.currentDrive = currentDrive
+                disableSelectedFolderId = disabledFolderId
+            }
 
-        currentFolderId?.let { folderId ->
-            // Simply navigate when the folder exists in the local database
-            FileController.getFileProxyById(folderId, customRealm = mainViewModel.realm)?.let {
-                initiateNavigationToCurrentFolder(folderId, currentUserDrive)
+            setSaveButton(customArgs)
+
+            currentFolderId?.let { folderId ->
+                // Simply navigate when the folder exists in the local database
+                FileController.getFileProxyById(
+                    fileId = folderId,
+                    userDrive = currentUserDrive,
+                    customRealm = getRealm(currentUserDrive)
+                )?.let {
+                    initiateNavigationToCurrentFolder(folderId, currentUserDrive)
+                }
             }
         }
     }
 
+    private fun getRealm(currentUserDrive: UserDrive) = if (currentUserDrive.sharedWithMe) null else mainViewModel.realm
```

Comment:
We should put that in a Utils somewhere, rename it to a more explicit name then use it in the other places where the same logic is used

Reply:
I'll move it to the ViewModel and put the `realm` as private to avoid other person to use this realm without the proper parameters.

Reply:
For now, we'll fix this only here (see with @sirambd ). We'll check other places in another PR if needed.